modules/text2vec-openai/vectorizer/objects.go

```diff
@@ -87,3 +115,181 @@ func (v *Vectorizer) getVectorizationConfig(cfg moduletools.ClassConfig) ent.Vec
 		Dimensions:   settings.Dimensions(),
 	}
 }
+
+// batchWorker is a go routine that handles the communication with the vectorizer
+//
+// On the high level it has the following steps:
+//  1. It receives a batch job
+//  2. It splits the job into smaller vectorizer-batches if the token limit is reached. Note that objects from different
+//     batches are not mixed with each other to simplify returning the vectors.
+//  3. It sends the smaller batches to the vectorizer
+func (v *Vectorizer) batchWorker() {
+	rateLimit := &ent.RateLimits{}
+	texts := make([]string, 0, 100)
+	firstRequest := true
+	timePerToken := 0.0
+	batchTookInS := float64(0)
+BatchLoop:
+	for job := range v.jobQueueCh {
+		// the total batch should not take longer than 60s to avoid timeouts. We will only use 40s here to be safe
+		batchStart := time.Now()
```

Comment:
if batch is supposed to be handled within `MaxBatchTime`, shouldn't `batchStart` be counted since object is pushed to the queue (so to be set on `batchJob` creation)? right now time spent in queue does not seem to be included.